packages/citation-js-utils/src/index.ts

```diff
@@ -252,13 +263,20 @@ export async function getCitationRenderers(data: CSL[]): Promise<CitationRendere
           },
           cite: c,
           getLabel(): string {
-            const bibtexObjects = cite.set(c).format('bibtex', { format: 'object' }) as {
-              label: string;
-            }[];
-            return bibtexObjects[0]?.label;
+            if (getLabel(cite, c).startsWith('temp_id_')) {
+              config.format.useIdAsLabel = false;
+            }
+            const label = getLabel(cite, c);
+            config.format.useIdAsLabel = true;
+            return label;
```

Comment:
Perhaps we should instead be normalising ingested entries so that we don't have to configure plugin config in a contextual manner?

Reply:
Yeah - I think you are right about getting rid of the contextual configuration.

We just need our own id generation for cases where the citation doesn't have an id.

Reply:
Ok - I just pulled out the `formatLabel` function from `citation-js` and used that in these cases, instead of the contextual configuration. Much better I think.